pkg/codegen/nodejs/gen.go

```diff
@@ -1422,9 +1515,11 @@ func (mod *modContext) getImportsForResource(member interface{}, externalImports
 				needsTypes = mod.getTypeImports(p.Type, false, externalImports, imports, seen) || needsTypes
 			}
 		}
-		if member.Outputs != nil {
-			for _, p := range member.Outputs.Properties {
-				needsTypes = mod.getTypeImports(p.Type, false, externalImports, imports, seen) || needsTypes
+		if member.ReturnType != nil {
+			if objectType, ok := member.ReturnType.(*schema.ObjectType); ok && objectType != nil {
+				for _, p := range objectType.Properties {
+					needsTypes = mod.getTypeImports(p.Type, false, externalImports, imports, seen) || needsTypes
+				}
```

Comment:
Is there a reason we only get imports for object types? The benefit of having `member.ReturnType` be a simple `schema.Type` is less special casing.
```suggestion
			needsTypes = mod.getTypeImports(member.ReturnType, false, externalImports, imports, seen) || needsTypes
```

Reply:
I refactored it like this so it doesn't change any old behaviour which relied on `Outputs` before but I guess it is worth simplifying now

Reply:
Actually applying this change results in the following redundant imports being added:
```ts
import * as inputs from "types/inputs";
import * as outputs from "types/inputs";
```
So leaving this one as is for now